File: 2024/March/3-25 442 Find All Duplicates In An Array/ArrayDuplicates.java
import java.util.List;
import java.util.ArrayList;

class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer> result = new ArrayList<Integer>();
        int index;
        for(int num : nums) {
            index = Math.abs(num) - 1;
            if(nums[index] < 0) {
                result.add(Math.abs(num));
            }
            else {
                nums[index] = -nums[index];
            }
        }

        return result;
    }
}

File: 2024/March/3-25 442 Find All Duplicates In An Array/ArrayDuplicates.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
    var result = Array();
    var index;

    for(var num of nums) {
        index = Math.abs(num) - 1;
        if(nums[index] < 0) {
            result.push(Math.abs(num));
        }
        else {
            nums[index] = -nums[index];
        }
    }

    return result;
};